sacc/tracers.py

```diff
@@ -198,13 +214,302 @@ def from_tables(cls, table_list):
+    @classmethod
+    def to_tables(cls, instance_list):
+        tables = []
+        for tracer in instance_list:
+            # Beams
+            names = ['ell', 'beam_ell']
+            cols = [tracer.ell, tracer.beam_ell]
+            for beam_id, col in tracer.beam_extra.items():
+                names.append(str(beam_id))
+                cols.append(col)
+            table = Table(data=cols, names=names)
+            table.meta['SACCTYPE'] = 'tracer'
+            table.meta['SACCCLSS'] = cls.tracer_type
+            table.meta['SACCNAME'] = tracer.name
+            table.meta['SACCQTTY'] = tracer.quantity
+            table.meta['EXTNAME'] = f'tracer:{cls.tracer_type}:{tracer.name}:beam'
+            table.meta['MAP_UNIT'] = tracer.map_unit
+            table.meta['SPIN'] = tracer.spin
+            for key, value in tracer.metadata.items():
+                table.meta['META_'+key] = value
+            remove_dict_null_values(table.meta)
+            tables.append(table)
+        return tables
+
+    @classmethod
+    def from_tables(cls, table_list):
+        tracers = {}
+
+        # Collect beam and bandpass tables describing the same tracer
+        tr_tables = {}
+        for table in table_list:
+            # Read name and table type
+            name = table.meta['SACCNAME']
+            quantity = table.meta['SACCQTTY']
+            tabtyp = table.meta['EXTNAME'].split(':')[-1]
+            if tabtyp not in ['beam']:
+                raise KeyError("Unknown table type " + table.meta['EXTNAME'])
+
+            # If not present yet, create new tracer entry
+            if name not in tr_tables:
+                tr_tables[name] = {}
+            # Add table
+            tr_tables[name][tabtyp] = table
+
+        # Now loop through different tracers and build them from their tables
+        for n, dt in tr_tables.items():
+            quantity = []
+            metadata = {}
+            map_unit = 'none'
+            ell = []
+            beam_ell = []
+            beam_extra = {}
+            spin = 0
+
+            if 'beam' in dt:
+                table = dt['beam']
+                name = table.meta['SACCNAME']
+                quantity = table.meta['SACCQTTY']
+                ell = table['ell']
+                beam_ell = table['beam_ell']
+                for col in table.columns.values():
+                    if col.name not in ['ell', 'beam_ell']:
+                        beam_extra[col.name] = col.data
+                map_unit = table.meta['MAP_UNIT']
+                spin = table.meta['SPIN']
+                for key, value in table.meta.items():
+                    if key.startswith("META_"):
+                        metadata[key[5:]] = value
+
+            tracers[name] = cls(name, quantity, spin, ell, beam_ell,
+                                beam_extra=beam_extra,
+                                map_unit=map_unit, metadata=metadata)
+        return tracers
+
+
+class NuMapTracer(BaseTracer, tracer_type='NuMap'):
```

Comment:
Why is this a whole new class as opposed to using the quantity data or something in the metadata?

Reply:
This type of tracer stores information about bandpass transmission and beams. This is ubiquitous enough that I think it deserves a separate class outside of `Misc`.